tests/Microsoft.Graph.DotnetCore.Core.Test/Requests/Middleware/RetryHandlerTests.cs

```diff
@@ -292,40 +295,45 @@ public async Task ShouldRetrytBasedOnRetryAfter(HttpStatusCode statusCode)
+        public async Task ShouldRetryBasedOnCustomShouldRetryDelegate(int expectedMaxRetry, HttpStatusCode expectedStatusCode)
+        {
+
+            var request = new HttpRequestMessage();
+            Queue<HttpResponseMessage> httpResponseQueue = new(new HttpResponseMessage[]
             {
-                Content = new StringContent("Hello World")
-            };
-            var retryResponse = new HttpResponseMessage(HttpStatusCode.BadGateway);
-
-            // create the retry response
-            var finalResponse = new HttpResponseMessage(HttpStatusCode.OK);
-
-            // hook everything up
-            this.testHttpMessageHandler.SetHttpResponse(retryResponse, finalResponse);
-
-            // Create a retry handler that will retry on HttpStatusCode.BadGateway and hook it up
-            RetryHandlerOption retryHandlerOption = new RetryHandlerOption
+                new(HttpStatusCode.BadGateway) { RequestMessage = request },
+                new(HttpStatusCode.BadGateway) { RequestMessage = request },
+                new(HttpStatusCode.BadGateway) { RequestMessage = request },
+                new(HttpStatusCode.BadGateway) { RequestMessage = request },
+                new(HttpStatusCode.OK) { RequestMessage = request },
+            });
+
+            var mockHttpMessageHandler = new Moq.Mock<HttpMessageHandler>(Moq.MockBehavior.Loose);
+            mockHttpMessageHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>())
+                .Returns(() => Task.FromResult(httpResponseQueue.TryDequeue(out HttpResponseMessage r) ? r : new(HttpStatusCode.OK) { RequestMessage = request }))
+                .Verifiable();
+
+            RetryHandler retryHandler = new(mockHttpMessageHandler.Object, new RetryHandlerOption()
             {
-                ShouldRetry = (delay, attempt, httpResponseMessage) => httpResponseMessage.StatusCode == HttpStatusCode.BadGateway
-            };
-            RetryHandler customHandler = new RetryHandler(this.testHttpMessageHandler, retryHandlerOption);
-            HttpMessageInvoker httpMessageInvoker = new HttpMessageInvoker(customHandler);
-
+                ShouldRetry = (delay, attempt, httpResponseMessage) => httpResponseMessage.StatusCode == HttpStatusCode.BadGateway,
+                MaxRetry = expectedMaxRetry,
+                Delay = 0
+            });
+
+            HttpMessageInvoker httpMessageInvoker = new(retryHandler);
+
             // Act
-            var response = await httpMessageInvoker.SendAsync(httpRequestMessage, new CancellationToken());
+            var response = await httpMessageInvoker.SendAsync(request, new CancellationToken());
 
             // Assert
-            Assert.Same(response, finalResponse);
-            Assert.NotSame(response.RequestMessage, httpRequestMessage);
-            Assert.NotNull(response.RequestMessage.Content);
```

Comment:
removing these checks, they really only cover the mock behavior of `testHttpMessageHandler` type to begin with. 

neither `retryHandler.SendAsync` nor `retryHandler.SendRetryAsync` have any conditionality or functionality that instantiates or modifies `HttpResponseMessage` nor `HttpResponseMessage.RequestMessage`.